### ISSUE TITLE
ArrayFormat Type Comma is producing the output of Array Format Repeat when the array length is 1

### ISSUE THREAD

**Issue by manoj-rp** [ljharb/qs#442]
`qs.stringify({ a: ['b'] }, { arrayFormat: 'comma' })` should return the `a=b`.

But it is returning the `repeat` type format - `a%5B%5D=b ~ a[]=b`

Below is the documentation and expected behaviour from readme
```
qs.stringify({ a: ['b', 'c'] }, { arrayFormat: 'indices' })
// 'a[0]=b&a[1]=c'
qs.stringify({ a: ['b', 'c'] }, { arrayFormat: 'brackets' })
// 'a[]=b&a[]=c'
qs.stringify({ a: ['b', 'c'] }, { arrayFormat: 'repeat' })
// 'a=b&a=c'
qs.stringify({ a: ['b', 'c'] }, { arrayFormat: 'comma' })
// 'a=b,c'
```
Below is the screenshot for the same

<img width="707" alt="image" src="https://user-images.githubusercontent.com/71893442/172765962-237e6d9f-dc65-4808-a084-87b93df95274.png">

Can anyone help me if this is an issue or Am I missing something?

**Comment by goodrick** [comment]
I encountered the same problem. I locked version 6.9.6

**Comment by manoj-rp** [comment]
Thanks for your reply. Unfortunately this module is being used by smoother module. And I'm using that module for my usecase.

**Comment by manoj-rp** [comment]
I have read the thread. But Shouldn't it be configurable?

Can we have another option along with **`arrayType`**. 

`preserveArrayness: true/false`

Because there can be usecases like 

One can use `qs` for stringifying it and may not be using it for parsing back again?


In our case,

We stringify it and pass the query string to an external service and that service doesn't use the `qs` module to parse it back?

Any thoughts @ljharb ljharb

**Comment by mima0815** [comment]
we run into the same issue with the new version.

not all APIs we are using are capable to handling query strings with brackets, they are expecting comma separated lists in all cases or no comma if it is only one value.

+1 for the additional option to toggle the "Arrayness" feature.

**Comment by glomotion** [comment]
we have also just slammed into this issue. Tis causing a great deal of errors in production for us right now (we use QS to stringify params which are sent to an API - which does NOT support array bracket query string notation). 😭

**Comment by Shooz136** [comment]
Same here - I'd appreciate a configurable option.

**Comment by ljharb** [comment]
I will plan on reverting the behavior by default, and adding an option to satisfy the original request.

**Comment by chengwb53** [comment]
突然莫名其妙出现这个问题，一对比，发现线上打包（自动部署）的代码和本地的代码不一致
![image](https://user-images.githubusercontent.com/18694507/174228469-18c2ae9f-fa53-4975-af10-20ac4714f1dd.png)
调试发现了这个问题，处理了单个的时候用 brackets

**Comment by lpillonel** [comment]
Same issue here.
If this change is intentional, I suggest to consider it as a breaking change

**Comment by ljharb** [comment]
Released in v6.11.0.
